numpyro/infer/elbo.py

```diff
@@ -960,24 +960,18 @@ def single_particle_elbo(rng_key):
                         eliminate=group_sum_vars | elim_plates,
                     )
                     # incorporate the effects of subsampling and handlers.scale through a common scale factor
-                    group_scales = {}
-                    for name in group_names:
-                        for plate, value in (
-                            model_trace[name].get("plate_to_scale", {}).items()
-                        ):
-                            if plate in group_scales:
-                                if value != group_scales[plate]:
-                                    raise ValueError(
-                                        "Expected all enumerated sample sites to share a common scale factor, "
-                                        f"but found different scales at plate('{plate}')."
-                                    )
-                            else:
-                                group_scales[plate] = value
-                    scale = (
-                        reduce(lambda a, b: a * b, group_scales.values())
-                        if group_scales
-                        else None
+                    scales_set = set(
+                        [
+                            model_trace[name]["scale"]
```

Comment:
I'm not sure how to check the uniqueness of jax array scales because jax arrays is not hashable. Could you replace this by something like
```
scale = None
for name in (group_names | group_sum_vars):
    site_scale = model_trace[name]["scale"]
    if isinstance(scale, (int, float)) and isinstance(site_scale, (int, float)) and (site_scale != scale):
        raise ValueError(...)
    scale = site_scale
```

Reply:
I copied the implementation from Pyro which first checks that array is scalar, converts it to scalar, and then compares them. Does it look alright to you?

https://github.com/pyro-ppl/pyro/blob/dev/pyro/infer/traceenum_elbo.py#L37-L41

> Btw, does this mean that we don't support enumeration for models with both global and local variables?

Yeah, if you enumerate a global variable than you cannot subsample a local variable that depends on it.
I'm working on it but that will require first changing `funsor.sum_product.sum_product` to handle plate-wise scaling there.

Reply:
Because we can't convert a tracer to float, `float(site_scale)` will fail when scale is a tracer. Maybe we need to raise error if it is a jnp.array?

Reply:
I see. Changed it so that it raises error if scale is `jnp.array`.